sql/server.go

```diff
@@ -175,6 +193,13 @@ func (s *Server) exec(req driver.Request) (resp driver.Response, err error) {
 		if err = gogoproto.Unmarshal(req.Session, &planMaker.session); err != nil {
 			return
 		}
+		if planMaker.session.Txn != nil {
+			planMaker.txn = s.db.ContinueTxn(*planMaker.session.Txn)
```

Comment:
The transaction state gets saved and sent back to the client. The client when reissuing a new command, sends the new command with the session state containing the transaction. Before running the command, the server reconstructs the transaction by calling ContinueTxn() and  issues the command in the transaction context.